SConstruct

```diff
@@ -1626,26 +1647,58 @@ if env['matlab_toolbox'] == 'y':
 # /usr/local because of dist-packages vs site-packages
 env['debian'] = any(name.endswith('dist-packages') for name in sys.path)
 
+# Check whether Cantera should be installed into a conda environment
+if conda_prefix is not None:
+    if env["layout"] != "conda":
+        use_conda = not selected_options & \
+            {"layout", "prefix", "python_prefix", "python_cmd"}
```

Comment:
Here, rather than the set intersection, I think you can just compare these values to their defaults, right?

Reply:
that's really tricky. I believe if a user specifies the default from the command line, this would not take effect? I believe it's safer to just read from `cantera.conf`, which is already done anyways for some of the logging output.

Reply:
> No, variables set on the command line are updated in `cantera.conf` on the `opts.Save` line, so they should be different from the default by this point. 

Correct. My implementation deliberately took advantage of this 'feature', as I believe *anything* specified either on the command line or in a pre-existing `cantera.conf` needs to take precedent. Unfortunately SCons doesn't expand variables in
```Python
opts = Variables("cantera.conf")
```
so I didn't see an obvious alternative solution. So overall, I stuck with this implementation as it reliably replicates all options.

Reply:
I can't get over that it somehow feels wrong or weird to be parsing the secondary level of configuration specification...

> Unfortunately SCons doesn't expand variables

I don't see how this is a problem? I assume you mean shell variables (like `$CONDA_PREFIX`). This is done by string operations somewhere around here (`os.path.expandvars` and `os.path.expanduser`).

If a variable is not specified in `cantera.conf` or on the command line (which are really equivalent), then they should still be equal to their default values as defined in the `Options` structure above. Checking whether the values for `layout`, `prefix`, `python_prefix`, and `python_cmd` are different from their default value (opts[value].default, I think...) should be equivalent and get directly to the data source that SCons is using to do its work.

For example, the default value of `python_prefix` is `"$prefix"` (as of this PR, #1158 changes that to the empty string, but that's not really important here).

Reply:
I’m not sure that we are disagreeing. The ‘secondary’ level succinctly contains what was specified? Even before this PR, this is exactly what was displayed for the summary.

The one case I am concerned a out is when a user specifies the the default value on the command line (or cantera.conf). For example, if you specify `standard` this would be equivalent to the default on some OSs, but would be overriden if conda is detected unless you query what was actually specified? You can probably come up with other scenarios where checking against the default could have unintended outcomes. Overall, I believe the current implementation does what it is supposed to?